File: leetcode/2996.大于等于顺序前缀和的最小缺失整数.go
func missingInteger(nums []int) int {
    // 将切片转换为集合
	s := make(map[int]bool)
	for _, num := range nums {
		s[num] = true
	}

	// 找到连续整数序列的最大值
	mxn := nums[0]
	for i := 1; i < len(nums); i++ {
		if nums[i] == nums[i-1]+1 {
			mxn += nums[i]
		} else {
			break
		}
	}

	// 找到缺失的整数
	for s[mxn] {
		mxn++
	}

	return mxn
}

File: leetcode/2996.大于等于顺序前缀和的最小缺失整数.py
class Solution:
    def missingInteger(self, nums: List[int]) -> int:
        s = set(nums)
        mxn = nums[0]
        for i in range(1,len(nums)):
            if nums[i] == nums[i-1]+1:
                mxn+=nums[i]
            else:
                break
            
        while mxn in s:
            mxn+=1
        return mxn